### PR TITLE
Add update task group

### PR DESCRIPTION
**Pull request by rbyshko**
Add the update task group - let me know if something is missing.

### CODE REVIEW

File: src/kraken/core/system/project.py [hunk 1]
```diff
@@ -71,6 +71,8 @@ def __init__(self, name: str, directory: Path, parent: Optional[Project], contex
         deploy_group = self.group("deploy", description="Tasks that deploy applications.")
         deploy_group.add_relationship(publish_group, strict=False)
 
+        update_group = self.group("update", description="Tasks that update the dependencies.")
```
Comment (tetigi) [inline src/kraken/core/system/project.py:74]: (very) minor nit: could we make the description `"Tasks that update dependencies of projects"`?

Comment (tetigi) [reply src/kraken/core/system/project.py:74]: You are also getting a flake8 fail due to `update_group` not being used. Make sure your `krakenw run lint` passes locally :)

Comment (NiklasRosenstein) [reply src/kraken/core/system/project.py:74]: `slap test` in this case :) and`slap run fmt`

**Comment by NiklasRosenstein** [comment]
> @NiklasRosenstein I wonder if explicit "update" group creation is needed here if it get's created dynamically anyway?

The only thing we gain by adding this is to give the group a description. 😄

**Comment by rbyshko** [comment]
Unfortunately `slap test` and `slap run fmt` are not consistent. For some reason `fmt` will not format `tests/test_resume_build_state.py`.

**Comment by rbyshko** [comment]
@NiklasRosenstein this is ready to merge - please review. Also please take a look at [this comment](https://github.com/kraken-build/kraken-core/pull/18#issuecomment-1425640812).

**Pull request merged**